pandas/core/arrays/datetimelike.py

```diff
@@ -2151,7 +2151,7 @@ def factorize(  # type:ignore[override]
 def ensure_arraylike_for_datetimelike(data, copy: bool, cls_name: str):
     if not hasattr(data, "dtype"):
         # e.g. list, tuple
-        if np.ndim(data) == 0:
+        if not isinstance(data, (list, tuple)) and np.ndim(data) == 0:
```

Comment:
Should `inspect.isgenerator` be used if we're specifically checking for a generator like in the comment?

Reply:
I'd keep as is to ensure that we don't miss anything. I think a white list is a bit better than only checking for generators